standards/code-style.adoc

```diff
@@ -338,12 +342,68 @@ if (condition) {
 }
 ----
 
-This is __bad__:
-
 [source, kotlin]
 ----
 if (condition)
-    body()  // bad!
+    body()  // WRONG!
+----
+
+Braces are not required for `when` branches and `if` statement bodies if they meet BOTH
+the following conditions:
+
+. the conditional and the branch each contain at most a single predicate
+. it fits on a single line
+
+For `if` statements, there is an additional condition that there be no `else if` or `else`
+branches.
+
+[source, kotlin]
+----
+// Okay
+if (string.isEmpty()) return
+
+// Okay
+when (value) {
+    0 -> doSpecificThing()
+    1 -> doAnotherSpecificThing()
+	else -> doGenericThing()
+}
+
+// WRONG
+if (string.isEmpty()) doEmptyThing() else doFullThing()
+
+// STILL WRONG!
+if (string.isEmpty()) doEmptyThing()
+else doFullThing()
+
+----
+
+==== Expressions
+
+An `if`/`else` conditional that is used as an expression may omit braces only if the
+conditional and each branch of the expression contain at most a single predicate,
+and the entire expression fits on one line.
+
+[source, kotlin]
+----
+val value = if (string.isEmpty()) 0 else 1  // Okay
```

Comment:
This actually seems like it might contradict the above statement I commented on. Am I missing something?

Reply:
One is an expression, one is a statement. Since Kotlin ditched the ternary operator in favor of this usage, it makes sense to allow it as an expression.

Reply:
Ahhh, okay, I see.